core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/login/LoginViewModel.kt

```diff
@@ -110,12 +110,21 @@ class LoginViewModel(
     private val parseCredentialUsernameUseCase: ParseCredentialUsernameUseCase by instance()
 
     init {
-        nextDestination = savedStateHandle[UstadView.ARG_NEXT] ?: ClazzListViewModel.DEST_NAME_HOME
 
         serverUrl = savedStateHandle[UstadView.ARG_LEARNINGSPACE_URL]
             ?: apiUrlConfig.presetLearningSpaceUrl ?: "http://localhost"
         savedStateHandle[UstadView.ARG_LEARNINGSPACE_URL] = serverUrl
 
+        val getDefaultDestinationUseCase: GetDefaultDestinationUseCase =
+            di.on(LearningSpace(serverUrl)).direct.instance()
+        viewModelScope.launch {
+            if (savedStateHandle[UstadView.ARG_NEXT].equals(PersonListViewModel.DEST_NAME_HOME)){
```

Comment:
I see no reason why PersonListViewModel needs special treatment here- this is a very weird if statement

Reply:
This is asynchronous code - and it is creating a race condition (automated test might go faster, and next destination would not be correct in that case). This needs to be refactored to avoid a race condition.